qa/restart-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/HybridSearchIT.java

```diff
@@ -0,0 +1,165 @@
+            addDocument(getIndexNameForTest(), "2", TEST_FIELD, TEXT_3, null, null);
+            addDocument(getIndexNameForTest(), "3", TEST_FIELD, TEXT_4, null, null);
+            addDocument(getIndexNameForTest(), "4", TEST_FIELD, TEXT_5, null, null);
+            createSearchPipeline(
+                SEARCH_PIPELINE1_NAME,
+                DEFAULT_NORMALIZATION_METHOD,
+                DEFAULT_COMBINATION_METHOD,
+                Map.of(PARAM_NAME_WEIGHTS, Arrays.toString(new float[] { 0.3f, 0.7f }))
+            );
+        } else {
+            Map<String, Object> pipeline = getIngestionPipeline(PIPELINE1_NAME);
+            assertNotNull(pipeline);
+            String modelId = getModelId(pipeline, TEXT_EMBEDDING_PROCESSOR);
+            loadModel(modelId);
+            addDocument(getIndexNameForTest(), "5", TEST_FIELD, TEXT_6, null, null);
+            validateTestIndex(modelId, getIndexNameForTest(), SEARCH_PIPELINE1_NAME);
+            deleteSearchPipeline(SEARCH_PIPELINE1_NAME);
+            deletePipeline(PIPELINE1_NAME);
+            deleteModel(modelId);
+            deleteIndex(getIndexNameForTest());
+        }
+    }
+
+    private String uploadTextEmbeddingModel() throws Exception {
+        String requestBody = Files.readString(Path.of(classLoader.getResource("processor/UploadModelRequestBody.json").toURI()));
+        return registerModelGroupAndGetModelId(requestBody);
+    }
+
+    private String registerModelGroupAndGetModelId(String requestBody) throws Exception {
+        String modelGroupRegisterRequestBody = Files.readString(
+            Path.of(classLoader.getResource("processor/CreateModelGroupRequestBody.json").toURI())
+        );
+        String modelGroupId = registerModelGroup(
+            String.format(LOCALE, modelGroupRegisterRequestBody, "public_model_" + RandomizedTest.randomAsciiAlphanumOfLength(8))
+        );
+        return uploadModel(String.format(LOCALE, requestBody, modelGroupId));
+    }
+
+    private void createPipelineProcessor(String modelId, String pipelineName) throws Exception {
+        String requestBody = Files.readString(Path.of(classLoader.getResource("processor/PipelineConfiguration.json").toURI()));
+        createPipelineProcessor(requestBody, pipelineName, modelId);
+    }
+
+    private void validateTestIndex(String modelId, String index, String searchPipeline) throws Exception {
+        int docCount = getDocCount(index);
+        assertEquals(6, docCount);
+        HybridQueryBuilder hybridQueryBuilder = getQueryBuilder(modelId);
+        Map<String, Object> searchResponseAsMap = search(index, hybridQueryBuilder, null, 1, Map.of("search_pipeline", searchPipeline));
+        assertNotNull(searchResponseAsMap);
+        int hits = getHitCount(searchResponseAsMap);
+        assertEquals(1, hits);
+        List<Double> scoresList = getNormalizationScoreList(searchResponseAsMap);
+        for (Double score : scoresList) {
+            assertTrue(0 < score && score < 1);
```

Comment:
However, I have simulated the same request present in the opensearch documentation so there I found the score to be between 0 and 1.

Reply:
can you please share the doc link. Score should be between 0 ands 1.0 for one sub-query and min/max technique. But in your case you have two sub-queries, if both giving you matches for same doc id then depending on how many docs we do have it can go as high as 2.0.

Reply:
```
 {
  "took": 64,
  "timed_out": false,
  "_shards": {
    "total": 3,
    "successful": 3,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": {
      "value": 3,
      "relation": "eq"
    },
    "max_score": 0.7,
    "hits": [
      {
        "_index": "neuralsearch-bwc-testnormalizationprocessor_whenindexwithmultipleshards_e2eflow",
        "_id": "1",
        "_score": 0.7,
        "_source": {
          "passage_embedding": [
            0.45678908,
            0.43721378,
            0.6018044,
            0.4522265,
            0.48888764,
            0.60348797,
            0.42700562,
            0.6021405,
            0.6524581,
            0.57113045,
            0.48201206,
            0.3457236,
            0.5172467,
            0.505291,
            0.49712038,
            0.653257,
            0.60487324,
            0.5016563,
            0.45953694,
            0.52661043,
            0.5699118,
            0.41417983,
            0.49392512,
            0.34313264,
            0.34226522,
            0.4323301,
            0.6062005,
            0.20702037,
            0.43142548,
            0.50183845,
            0.58372265,
            0.42687285,
            0.48335278,
            0.5340483,
            0.43873703,
            0.42726776,
            0.4510444,
            0.44701284,
            0.48849374,
            0.4617867,
            0.53596854,
            0.41940582,
            0.64726883,
            0.5061251,
            0.35471612,
            0.5495474,
            0.55167055,
            0.5297058,
            0.4504541,
            0.5201292,
            0.49827173,
            0.66357684,
            0.37060732,
            0.38965186,
            0.6080643,
            0.33265236,
            0.42078027,
            0.40636122,
            0.57270664,
            0.5298725,
            0.55376965,
            0.4908773,
            0.50343484,
            0.49225026,
            0.5640268,
            0.5027385,
            0.557072,
            0.3740399,
            0.5451468,
            0.5830458,
            0.40049034,
            0.4375475,
            0.56251395,
            0.3870018,
            0.48882753,
            0.45834053,
            0.46088892,
            0.6317374,
            0.3728965,
            0.40242806,
            0.5332909,
            0.5966067,
            0.456397,
            0.45135456,
            0.72924346,
            0.5786153,
            0.5873974,
            0.5042727,
            0.56993365,
            0.5222905,
            0.5744427,
            0.48224992,
            0.52719986,
            0.3933128,
            0.55830044,
            0.40232855,
            0.65540874,
            0.4859558,
            0.46116352,
            0.43577456,
            0.47170445,
            0.52465236,
            0.39901862,
            0.47643986,
            0.47745144,
            0.42665842,
            0.40262565,
            0.43875584,
            0.4022803,
            0.61152214,
            0.62895685,
            0.5946106,
            0.44095728,
            0.44802603,
            0.51158404,
            0.61185795,
            0.5578354,
            0.54729,
            0.4399002,
            0.46155587,
            0.39209798,
            0.4625926,
            0.33531487,
            0.61760014,
            0.52370244,
            0.59759635,
            0.38223633,
            0.46970236,
            0.449347,
            0.6388946,
            0.4571052,
            0.43812656,
            0.6602506,
            0.5212066,
            0.5984724,
            0.41208038,
            0.6579074,
            0.4126607,
            0.4292409,
            0.5517119,
            0.6674939,
            0.31386766,
            0.49233958,
            0.31372288,
            0.5754148,
            0.6831956,
            0.6033165,
            0.6202802,
            0.43602908,
            0.2806899,
            0.4029851,
            0.5299538,
            0.53667724,
            0.5170701,
            0.46559602,
            0.5981658,
            0.55689514,
            0.35027406,
            0.5819456,
            0.49300534,
            0.4174732,
            0.46207336,
            0.57241595,
            0.59198403,
            0.35277888,
            0.43361336,
            0.45903662,
            0.4691988,
            0.5417924,
            0.6412136,
            0.4820869,
            0.47528225,
            0.26110482,
            0.49012232,
            0.30506834,
            0.4440388,
            0.53342974,
            0.5026627,
            0.40252638,
            0.49858114,
            0.41736364,
            0.6119264,
            0.5186521,
            0.47745204,
            0.60754097,
            0.6677315,
            0.44689086,
            0.5103237,
            0.6417963,
            0.5032117,
            0.41029415,
            0.5033442,
            0.52525294,
            0.59287745,
            0.49268675,
            0.61570895,
            0.52425677,
            0.37308905,
            0.66980773,
            0.4342157,
            0.49409613,
            0.35559377,
            0.37215158,
            0.46626255,
            0.41426468,
            0.3568931,
            0.37906674,
            0.5895227,
            0.43388546,
            0.5456163,
            0.5239831,
            0.4734403,
            0.40124077,
            0.4230839,
            0.41635275,
            0.5798791,
            0.52363735,
            0.5893671,
            0.29025692,
            0.57402,
            0.41584226,
            0.4962776,
            0.36189103,
            0.5843178,
            0.44928113,
            0.5636265,
            0.5346425,
            0.5247843,
            0.48535502,
            0.45527664,
            0.41098073,
            0.36051387,
            0.5201383,
            0.4915119,
            0.62943006,
            0.5186017,
            0.42780703,
            0.55647373,
            0.4987392,
            0.5893299,
            0.55515486,
            0.56412244,
            0.42849934,
            0.5837349,
            0.56887317,
            0.35389516,
            0.6025011,
            0.5896341,
            0.52403927,
            0.4531467,
            0.36720982,
            0.6176794,
            0.51780826,
            0.6424091,
            0.48861757,
            0.3459044,
            0.5149392,
            0.7277246,
            0.3618381,
            0.3509781,
            0.5217694,
            0.6108825,
            0.50911736,
            0.47885922,
            0.5280787,
            0.30983698,
            0.68921566,
            0.61406,
            0.56451637,
            0.54423773,
            0.61455506,
            0.44073847,
            0.5569231,
            0.36377242,
            0.61174834,
            0.41504726,
            0.64561474,
            0.41993445,
            0.457118,
            0.38874996,
            0.5625628,
            0.44353136,
            0.5879026,
            0.5949569,
            0.5131908,
            0.49310467,
            0.6285291,
            0.578754,
            0.45332292,
            0.5680776,
            0.3373917,
            0.402712,
            0.40476254,
            0.56656134,
            0.50204784,
            0.5964773,
            0.47512192,
            0.6116282,
            0.4550452,
            0.52948874,
            0.5660101,
            0.4043285,
            0.2974824,
            0.47465307,
            0.6033621,
            0.40385967,
            0.4734995,
            0.2712129,
            0.5246625,
            0.49144313,
            0.72023624,
            0.42113954,
            0.6658187,
            0.43771812,
            0.5340698,
            0.44211978,
            0.61998135,
            0.45269552,
            0.47757506,
            0.62823147,
            0.54207826,
            0.6939408,
            0.34883568,
            0.54311067,
            0.56407714,
            0.48074523,
            0.31684428,
            0.4406764,
            0.46164614,
            0.6027986,
            0.46669707,
            0.55378705,
            0.53739953,
            0.3969934,
            0.55988914,
            0.54608524,
            0.4688216,
            0.6652522,
            0.41394648,
            0.45485586,
            0.44504082,
            0.40830058,
            0.59447247,
            0.5183436,
            0.46410716,
            0.6236215,
            0.5362633,
            0.50564116,
            0.54710764,
            0.37946784,
            0.5860443,
            0.5820471,
            0.57767856,
            0.47966066,
            0.51445913,
            0.4607857,
            0.47769746,
            0.61964834,
            0.56345516,
            0.45972583,
            0.64311075,
            0.39685085,
            0.4742454,
            0.5407227,
            0.380226,
            0.32656908,
            0.43757465,
            0.6841548,
            0.5090351,
            0.34499457,
            0.42987198,
            0.60157126,
            0.5567616,
            0.5662084,
            0.40513128,
            0.43598577,
            0.411141,
            0.4809225,
            0.680317,
            0.7347916,
            0.42602658,
            0.42337233,
            0.55613923,
            0.4176475,
            0.50904477,
            0.59965247,
            0.68118775,
            0.45607007,
            0.5966442,
            0.55025107,
            0.48150286,
            0.6197405,
            0.40251434,
            0.47262728,
            0.5396453,
            0.45568773,
            0.56072617,
            0.38685483,
            0.51066935,
            0.42002675,
            0.68335456,
            0.5274244,
            0.6410165,
            0.43417615,
            0.5004573,
            0.4165367,
            0.4959835,
            0.6174581,
            0.57161415,
            0.5605112,
            0.5394618,
            0.42863622,
            0.3880745,
            0.5764023,
            0.4103215,
            0.4761246,
            0.5057282,
            0.43961787,
            0.47638407,
            0.4439488,
            0.4354511,
            0.36528215,
            0.42863172,
            0.49905527,
            0.46250007,
            0.55528474,
            0.27748448,
            0.38050655,
            0.49875772,
            0.37370592,
            0.50715035,
            0.4179396,
            0.535996,
            0.5053005,
            0.49288645,
            0.46087232,
            0.39640126,
            0.57829416,
            0.45362964,
            0.59966826,
            0.52998006,
            0.537942,
            0.622216,
            0.55185115,
            0.49129853,
            0.51418096,
            0.5732721,
            0.48292923,
            0.5661372,
            0.6737421,
            0.43246096,
            0.5232238,
            0.60964334,
            0.41778287,
            0.41623968,
            0.5971152,
            0.42939416,
            0.45100614,
            0.40751705,
            0.42347035,
            0.50785553,
            0.66719085,
            0.50423396,
            0.6969637,
            0.3887195,
            0.46817285,
            0.7143599,
            0.5947203,
            0.4604264,
            0.59032565,
            0.4903084,
            0.46937126,
            0.4175969,
            0.45003894,
            0.5323193,
            0.494877,
            0.6331888,
            0.6578008,
            0.5857233,
            0.5302454,
            0.5116009,
            0.47048688,
            0.40209815,
            0.5951693,
            0.64942193,
            0.57206553,
            0.53599274,
            0.3075231,
            0.58966905,
            0.4953017,
            0.5202224,
            0.5707398,
            0.5936642,
            0.42258653,
            0.5722946,
            0.6352535,
            0.5294653,
            0.4554614,
            0.5686836,
            0.4923945,
            0.27553886,
            0.41030324,
            0.39819998,
            0.50330955,
            0.68946564,
            0.4704165,
            0.39218712,
            0.6472176,
            0.6068535,
            0.43671188,
            0.5159077,
            0.7288941,
            0.6244533,
            0.5542591,
            0.5422273,
            0.5864541,
            0.58690625,
            0.50648063,
            0.5026035,
            0.5104286,
            0.5886998,
            0.45886102,
            0.5547869,
            0.6155639,
            0.5098352,
            0.4987538,
            0.5005282,
            0.5564267,
            0.6709813,
            0.41267902,
            0.45903277,
            0.28421196,
            0.52120924,
            0.5144159,
            0.6046032,
            0.5322981,
            0.53203547,
            0.5274364,
            0.6078127,
            0.44592467,
            0.6004222,
            0.56916535,
            0.5683681,
            0.5006443,
            0.5292486,
            0.43402904,
            0.55341995,
            0.44262108,
            0.507177,
            0.40674713,
            0.4693887,
            0.50978637,
            0.39800063,
            0.4790139,
            0.59340173,
            0.60026735,
            0.56911576,
            0.7194923,
            0.36878508,
            0.56234086,
            0.48649755,
            0.47397164,
            0.4364098,
            0.5811409,
            0.455487,
            0.54617614,
            0.5294583,
            0.5764207,
            0.50414056,
            0.6181972,
            0.33615735,
            0.26753482,
            0.44644552,
            0.41151977,
            0.54726934,
            0.47384137,
            0.55689377,
            0.6771259,
            0.3397008,
            0.49687627,
            0.67361397,
            0.45784622,
            0.381784,
            0.4547865,
            0.54370904,
            0.5563656,
            0.44377464,
            0.4626493,
            0.65625507,
            0.5189786,
            0.52697635,
            0.47224537,
            0.60353404,
            0.618739,
            0.51862216,
            0.5580649,
            0.49110225,
            0.43977156,
            0.5692691,
            0.47953942,
            0.52731025,
            0.67416465,
            0.51501155,
            0.3925893,
            0.43122938,
            0.47125077,
            0.5824047,
            0.46686438,
            0.5835651,
            0.364848,
            0.59473115,
            0.41757232,
            0.5071991,
            0.51134473,
            0.40976182,
            0.48523748,
            0.47457358,
            0.42035177,
            0.40012127,
            0.5993925,
            0.60178286,
            0.27026337,
            0.59058154,
            0.45672438,
            0.54370075,
            0.504732,
            0.57129884,
            0.581653,
            0.60007775,
            0.42897296,
            0.37106663,
            0.52296704,
            0.5671765,
            0.59980714,
            0.5454429,
            0.5315913,
            0.61733794,
            0.43377867,
            0.40146738,
            0.38577038,
            0.519411,
            0.4007752,
            0.61756504,
            0.6407042,
            0.52958447,
            0.50036365,
            0.36452857,
            0.38137084,
            0.30597624,
            0.51858854,
            0.32573777,
            0.5185772,
            0.47016296,
            0.49555838,
            0.40051305,
            0.60270876,
            0.5554788,
            0.5026391,
            0.53227174,
            0.5333888,
            0.5074061,
            0.32004097,
            0.4234308,
            0.44039357,
            0.43878356,
            0.51928365,
            0.5181041,
            0.40979803,
            0.5476924,
            0.4356667,
            0.4887812,
            0.42941287,
            0.45691448,
            0.46338928,
            0.5517,
            0.33034894,
            0.6072024,
            0.4967248,
            0.4075187,
            0.34948957,
            0.46437615,
            0.48490834,
            0.52303636,
            0.41481012,
            0.50513965,
            0.49018258,
            0.39225414,
            0.36277986,
            0.55263823,
            0.5010322,
            0.5671586,
            0.4795716,
            0.44340083,
            0.55455875,
            0.51574445,
            0.572265,
            0.33450362,
            0.5453222,
            0.55384696,
            0.45050105,
            0.7384049,
            0.57742876,
            0.5095947,
            0.5192233,
            0.53580046,
            0.38307762,
            0.57252586,
            0.42517778,
            0.36563432,
            0.6331501,
            0.5351396,
            0.33045292,
            0.4145699,
            0.624864,
            0.412383,
            0.49449807,
            0.6704041,
            0.38699624,
            0.33316386,
            0.46175075,
            0.5021057,
            0.45468524,
            0.47050938,
            0.42652154,
            0.56900173,
            0.40316698,
            0.54877436,
            0.74015075,
            0.34851262,
            0.4991066,
            0.47411093,
            0.44044796,
            0.46466085,
            0.3653556,
            0.4328874,
            0.62103975,
            0.63522464,
            0.3430772,
            0.27373916,
            0.51816154,
            0.37923428,
            0.43010074,
            0.3403915,
            0.6131996,
            0.44975036,
            0.42400903,
            0.56415623,
            0.4884729,
            0.49108374,
            0.48579004,
            0.3613526,
            0.41554224,
            0.3898419,
            0.3775853,
            0.48518378,
            0.44468608,
            0.5351658,
            0.64840555,
            0.615056,
            0.57635856,
            0.40089804,
            0.6745618,
            0.42559576,
            0.3749626,
            0.6317024,
            0.59512585,
            0.39228687,
            0.38443306
          ],
          "passage_text": "Hi planet"
        }
      }
    ]
  }
}

```

Reply:
it will work in most cases, but not always, depending on number of shards, sub-queries and doc relative position for each sub-query. Your method for index validation looks more like generic one. If you going to call it for multiple scenarios then logic should be generic too.